### PR TITLE
group by operator

### PR DESCRIPTION
**Pull request by drunkirishcoder**

### CODE REVIEW

File: framework/src/new_operators/mod.rs [hunk 1]
```diff
@@ -114,21 +147,65 @@ pub mod tests {
...
+                .group_by(
+                    2,
+                    |p| match p.protocol() {
+                        ProtocolNumbers::Tcp => 0,
+                        ProtocolNumbers::Udp => 1,
+                        _ => 2
+                    },
+                    |mut groups| {
+                        merge![
+                            groups.remove(&0)
```
Comment (zeeshanlakhani) [inline framework/src/new_operators/mod.rs]: I think we discussed wrapping these up as they seem odd in the API/interface sense.

File: test/ipv4or6/src/main.rs [hunk 2]
```diff
@@ -24,7 +26,25 @@ where
 
     let pipelines: Vec<_> = ports
         .iter()
-        .map(|port| tcp_nf(ReceiveBatch::new(port.clone()), sched).send(port.clone()))
+        .map(|port| {
```
Comment (zeeshanlakhani) [inline test/ipv4or6/src/main.rs]: I think we also discussed that for these "small" NB functional tests, we should just use `main.rs` to organize functions and the main call site together. We can revisit separation upon updating the current OG NF.

File: framework/src/new_operators/queue_batch.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
+use super::{Batch, PacketError};
```
Comment (zeeshanlakhani) [inline framework/src/new_operators/queue_batch.rs]: We should probably have some module documentation here.

File: framework/src/new_operators/mod.rs (hunk 1 above)
Comment (zeeshanlakhani) [reply framework/src/new_operators/mod.rs]: know you're on it @drunkirishcoder

File: framework/src/new_operators/queue_batch.rs (hunk 3 above)
Comment (zeeshanlakhani) [reply framework/src/new_operators/queue_batch.rs]: @drunkirishcoder just sayin :).

Comment (drunkirishcoder) [reply framework/src/new_operators/queue_batch.rs]: I will add more doc in the next PR where I'm expanding on the queue batch functionality with packet generation.

Comment (zeeshanlakhani) [reply framework/src/new_operators/queue_batch.rs]: good deal.

**Pull request merged**